docs/Dialects/Comb/RationaleComb.md

```diff
@@ -177,6 +183,24 @@ We agreed that we'd revisit in the future if there were a specific reason to
 add it.  Until then we represent the `array_create`/`array_get` pattern for
 frontends that want to generate this.
 
+
+### Undefined behavior for division
+
+`divu` and `divs` have undefined behavior when the numerator is 0.  It is 
+expected that a frontend will use additional operations to implement the 
+semantics required for that language.  For example, system verilog returns an 
+`x` on divide by zero, thus its representation may be
+`mux(denominator == 0, sv.constantx, divu(numerator,denominator))` whereas VHDL
+has a runtime-trap in simulation, thus it may require 
+`if(denominator==0) { assert() } else { divu(numerator,denominator)} ` and
+FIRRTL requires division by zero to return 0, so it may require
+`mux(denominator==0, 0, divu(numerator,denominator))`.
```

Comment:
FIRRTL (the spec) doesn't require this. FIRRTL compilers have historically been forced to implement exactly two division optimizations: `div(a, a) -> 1` and `div(a, 1) -> a`. This was due to the SFC implementing these optimizations and then MFC needing this to match during the transition. See discussion: https://github.com/llvm/circt/pull/405

Today, CIRCT will emit a bare divide by zero if you provide it:

``` firrtl
FIRRTL version 4.0.0
circuit Foo:
  public module Foo:
    input a: UInt<2>
    output b: UInt<2>

    connect b, div(a, UInt<2>(0))
```

Produces:

``` verilog
// Generated by CIRCT firtool-1.73.0-50-gef689399c
module Foo(
  input  [1:0] a,
  output [1:0] b
);

  assign b = a / 2'h0;
endmodule
```